src/main/plugin/iso19139.ca.HNAP/process/onlinesrc-add.xsl

```diff
@@ -371,9 +374,12 @@ Insert is made in first transferOptions found.
           <gmd:onLine>
             <xsl:variable name="isMapProtocol" select="starts-with($protocol, 'ESRI REST:') or starts-with($protocol, 'OGC:')" />
 
+            <xsl:variable name="descMainLang" select="substring-before($desc, '|')" />
+            <xsl:variable name="descTranslatedLang" select="substring-after($desc, '|')" />
+
             <xsl:variable name="resourceLang">
               <xsl:choose>
-                <xsl:when test="ends-with($desc, 'fra')"><xsl:value-of select="'urn:xml:lang:fra-CAN'"/></xsl:when>
+                <xsl:when test="ends-with($descMainLang, 'fra') and ends-with($descTranslatedLang, 'fra')"><xsl:value-of select="'urn:xml:lang:fra-CAN'"/></xsl:when>
```

Comment:
I would expect that we would only really care about $descMainLang. If $descTranslatedLang is not correct then it is a translation issue.

Otherwise, in the case where someone messes up the translation

`desc: eng#Web Service;ESRI REST;fra|fra#Service Web;ESRI REST;eng`

In this case I think it is better to use `urn:xml:lang:fra-CAN` as identified in the main language than to default to `urn:xml:lang:eng-CAN` because the match failed.

Also we should probably be careful for the following case.
`desc: eng#Web Service;ESRI REST;eng,fra|fra#Service Web;ESRI REST;eng`

In this case it ends with fra but it is not French, it is bilingual. In this case - I'm not sure what it best to use. But it does not seem like French is the best options.  But nor is English.  I wonder if if the last parameter is not "eng" or "fra" if it would be better to default to no value for resourceLang as we don't know.

So in this case, we should not check for 
`substring-before($desc, '|')`
But 
`<xsl:variable name="descMainLang" select="tokenize(substring-before($desc, '|'), ';')[last()]" />`

Then instead of doing 
`ends-with($descMainLang, 'fra')`

you could do `$descMainLang = 'fra'` and you would also need a case for `$descMainLang = 'eng'`

Then the default would be nothing.
 - eng=urn:xml:lang:fra-CAN
 - fra=urn:xml:lang:eng-CAN
 - eng,fra=null
 
 I'm not sure if this would make more sense? - Not sure if the last case of setting it to null will cause issues?

Reply:
I have made change change to parse the better descMainLang 

But as for the default language setup part, I am hesitate to do it because we haven't studied very well into if the language code of null will impact other things. My personal opinion will just keep the original logic for checking end-with. We could open a new issue or PR to track this specific issue and do some further investigation. But this PR is to mainly fix the validation inconsistent.